e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1

```diff
@@ -0,0 +1,477 @@
+###########################################################################
+
+Function Connect-AzureSubscription()
+{
+    # Ensure the user is logged in
+    try
+    {
+        $azureContext = az account show
+    }
+    catch {
+    }
+
+    if (-not $azureContext)
+    {
+        Write-Host "`nPlease login to Azure..."
+        az login
+        $azureContext = az account show
+    }
+
+    # Ensure the desired subscription is selected
+    if ($azureContext.id -ne $SubscriptionId)
+    {
+        Write-Host "`nSelecting subscription $SubscriptionId"
+        az account set --subscription $SubscriptionId
+    }
+
+    return $azureContext
+}
+
+#################################################################################################
+# Set required parameters
+#################################################################################################
+
+$Region = $Region.Replace(' ', '')
+
+$StorageAccountName = "$($ResourceGroup.ToLower())sa"
+$HubName = $ResourceGroup
+$AppRegistrationName = $ResourceGroup
+$DeviceProvisioningServiceName = $ResourceGroup
+$FarRegion = "southeastasia"
+$FarHubName = $ResourceGroup + "Far"
+$UploadCertificateName = "group1-certificate"
+
+############################################################################################################################
+# Cleanup old certs and files that can cause a conflict
+############################################################################################################################
+
+Write-Host("`nCleanup old certs and files that can cause a conflict")
+Get-ChildItem "Cert:\LocalMachine\My" | Where-Object {$_.Issuer.Contains("CN=$SubjectPrefix") } | Remove-Item
+Get-ChildItem "Cert:\LocalMachine\My" | Where-Object {$_.Issuer.Contains("CN=xdevice1") } | Remove-Item
+Get-ChildItem "Cert:\LocalMachine\My" | Where-Object {$_.Issuer.Contains("CN=iothubx509device1") } | Remove-Item
+Get-ChildItem .\ | Where-Object  {$_.Name.EndsWith(".pfx") } | Remove-Item
+Get-ChildItem .\ | Where-Object  {$_.Name.EndsWith(".cer") } | Remove-Item
+Get-ChildItem .\ | Where-Object  {$_.Name.EndsWith(".p7b") } | Remove-Item
+
+########################################################################################################
+# Generate self signed certs and to use in DPS and IOT Hub
+# New certs will be generated each time you run the script as the script cleans up in the end
+########################################################################################################
+
+Write-Host "`nGenerating self singed certs"
+
+$SubjectPrefix = "Azure IoT Test"
+$RootCommonName = "$SubjectPrefix Test Root CA"
+$IntermediateCert1CommonName = "$SubjectPrefix Intermediate 1 CA"
+$IntermediateCert2CommonName = "$SubjectPrefix Intermediate 2 CA"
+$GroupCertCommonName = "xdevice1"
+$DeviceCertCommonName = "iothubx509device1"
+$IotHubCertCommonName = "iothubx509device1"
+
+$RootCertPath = "./Root.cer"
+$IndividualDeviceCertPath = "./Device.cer"
```

Comment:
Some variables are TitleCase and others are camelCase.

Script params should the the only TitleCase ones, right?

Reply:
yeah I guess so. I can fix that.